rtgui/editorpanel.cc

```diff
@@ -1885,6 +1893,30 @@ void EditorPanel::sendToGimpPressed ()
 }
 
 
+bool EditorPanel::saveImmediately(const Glib::ustring &filename, SaveFormat sf)
+{
+    rtengine::procparams::ProcParams pparams;
+    ipc->getParams (&pparams);
+    std::unique_ptr<rtengine::ProcessingJob> job(rtengine::ProcessingJob::create (ipc->getInitialImage(), pparams));
+
+    // save immediately
+    rtengine::IImage16 *img = rtengine::processImage(job.get(), err, nullptr, options.tunnelMetaData, false);
```

Comment:
@agriggio I like your use of a `std::unique_ptr` here. :+1: There's a handful of other places calling `rtengine::processImage()` where the `job` instance isn't freed after use. Maybe you could sanitize them, too?